docs/book/developer-guide/advanced-usage/secret-references.md

```diff
@@ -0,0 +1,50 @@
+---
+description: How to reference secrets when configuring stack components
+---
+
+Some of the components in your stack require you to configure them with 
+sensitive information like passwords or tokens so they can connect to the 
+underlying infrastructure. Secret references allow you to configure these components in
+a secure way by not specifying the value directly but instead referencing a secret.
+To reference a secret in any string attribute of your stack components, simply specify
+the attribute value using the following syntax:
+```shell
+{{<SECRET_NAME>.<SECRET_KEY>}}
```

Comment:
how does `{SECRET_NAME}` interact with scoped secrets? It's not clear to me from just the documentation if this is scoped by default or not or how you might override the scope

Reply:
Secret scoping is currently configured on the secrets manager and handled in its internal implementation. In that sense, the scoping of this is tied to the scope of the secrets manager of the active stack.
For example: `{{secret.key}}` with a namespace-scoped secrets manager will fetch the secret `zenml/namespace/<NAMESPACE>/secret`
I don't think our secrets managers currently allow an easy way of overriding the scope, maybe @stefannica has some opinion on that?

Reply:
@strangemonad the `{SECRET_NAME}` here is independent of the secret scope. Whatever your current secrets manager is scoped to, that's what will be used to resolve it.